pybossa/view/projects.py

```diff
@@ -2925,3 +2925,34 @@ def assign_users(short_name):
 
     flash(msg, 'success')
     return redirect_content_type(url_for('.settings', short_name=project.short_name))
+
+@blueprint.route('/<short_name>/quiz-mode', methods=['GET', 'POST'])
+def quiz_mode(short_name):
+    project = project_repo.get_by_shortname(short_name)
+
+    ensure_authorized_to('read', project)
+    ensure_authorized_to('update', project)
+
+    if request.method == 'POST':
+        form = ProjectQuizForm(request.form)
+        if not form.validate():
+            flash("Please fix the errors", 'message')
+        else:
+            field_names = [
+                'enabled',
+                'questions_per_quiz',
+                'correct_answers_to_pass'
+            ]
+            project.info['quiz'] = { field_name: getattr(form, field_name).data for field_name in field_names }
+            project_repo.update(project)
```

Comment:
as its project config update, success can be audited and errors reported under logs.

Reply:
For the error logging what should I log? I'm not handling any specific errors in this function.

Reply:
basically anything info/error that would help troubleshooting can be logged. in this case, if raising validation error on correct ans > total is already getting logged and there's no other specific info/error to log, its fine to ignore logging here.